src/foam/core/ClassLoader.js

```diff
@@ -17,48 +17,92 @@
 
 foam.CLASS({
   package: 'foam.core',
+  name: 'ModelARequireExtension',
+  refines: 'foam.core.Model',
+
+  methods: [
+    function arequire() {
+      var p = Promise.resolve();
+      if (this.extends) {
+        p = p.then(this.__subContext__.arequire(this.extends));
+      }
+      for (var i = 0, a; a = this.axioms_[i]; i++) {
+        if (a.arequire) {
+          p = p.then(a.arequire());
```

Comment:
I don't believe this is correct.  You can't pass a promise to p.then().

What you really want is to either build an array of promises

promises.push(a.arequire());

and then do return Promise.all(promises);

In fact I would restructure this whole method as such

var promises = [];

if ( this.extends ) promises.push(this.__subContext__.arequire(this.extends));

for ( var i = 0, a; a = this.axioms... ) {
  if ( a.arequire ) promises.push(a.arequire());
}

return Promise.all(promises);